From: Michal Hocko <mhocko@suse.cz>
Subject: kabi: invalidate removed sys_elem_dir::children
Patch-mainline: never, kabi fixup
References: bnc#919589

sys_elem_dir::children has been removed from the code but we have to preserve
it to preserve the kabi (size of the structure). Let's change the type to void*
to catch a potential abuse in compile time.

Signed-off-by: Michal Hocko <mhocko@suse.cz>

---
 fs/sysfs/dir.c   |    1 +
 fs/sysfs/mount.c |    1 +
 fs/sysfs/sysfs.h |    5 +++++
 3 files changed, 7 insertions(+)

--- a/fs/sysfs/dir.c
+++ b/fs/sysfs/dir.c
@@ -22,6 +22,7 @@
 #include <linux/mutex.h>
 #include <linux/slab.h>
 #include <linux/security.h>
+#include <linux/poison.h>
 #include "sysfs.h"
 
 DEFINE_MUTEX(sysfs_mutex);
--- a/fs/sysfs/mount.c
+++ b/fs/sysfs/mount.c
@@ -19,6 +19,7 @@
 #include <linux/module.h>
 #include <linux/magic.h>
 #include <linux/slab.h>
+#include <linux/poison.h>
 
 #include "sysfs.h"
 
--- a/fs/sysfs/sysfs.h
+++ b/fs/sysfs/sysfs.h
@@ -19,7 +19,12 @@ struct sysfs_open_dirent;
 struct sysfs_elem_dir {
 	struct kobject		*kobj;
 
+#ifdef __GENKSYMS__
 	struct sysfs_dirent	*children;	/* No longer used */
+#else
+	/* Make sure the compilation fails if somebody still needs it */
+	void			*children_unused;
+#endif
 
 	unsigned long		subdirs;
 };
